scikeras/utils/transformers.py

```diff
@@ -0,0 +1,201 @@
+        self, X: np.ndarray, return_proba: bool = False
+    ) -> np.ndarray:
+        y = X  # rename for clarity, the input is always expected to be a target `y`
+        if self._target_type == "binary":
+            # array([0.9, 0.1], [.2, .8]) -> array(['yes', 'no'])
+            if self.n_classes_ == 1:
+                # special case: single input label for sigmoid output
+                # may give more predicted classes than inputs for
+                # small sample sizes!
+                # don't even bother inverse transforming, just fill.
+                class_predictions = np.full(
+                    shape=(y.shape[0], 1), fill_value=self.classes_[0]
+                )
+            else:
+                if y.shape == 1 or (y.shape[1] == 1 and self.n_classes_ == 2):
+                    # result from a single sigmoid output
+                    # reformat so that we have 2 columns
+                    y = np.column_stack([1 - y, y])
+                y_ = np.argmax(y, axis=1).reshape(-1, 1)
+                class_predictions = self.final_encoder_.inverse_transform(y_)
+        elif self._target_type == "multiclass":
+            # array([0.8, 0.1, 0.1], [.1, .8, .1]) ->
+            # array(['apple', 'orange'])
+            idx = np.argmax(y, axis=-1)
+            if not is_categorical_crossentropy(self.loss):
+                y_ = idx.reshape(-1, 1)
+            else:
+                y_ = np.zeros(y.shape, dtype=int)
+                y_[np.arange(y.shape[0]), idx] = 1
+            class_predictions = self.final_encoder_.inverse_transform(y_)
+        elif self._target_type == "multiclass-onehot":
+            # array([.8, .1, .1], [.1, .8, .1]) ->
+            # array([[1, 0, 0], [0, 1, 0]])
+            idx = np.argmax(y, axis=-1)
+            y_ = np.zeros(y.shape, dtype=int)
+            y_[np.arange(y.shape[0]), idx] = 1
+            class_predictions = y_
+        else:
+            class_predictions = None
+
+        if return_proba:
+            return y
+        return np.squeeze(np.column_stack(class_predictions)).astype(
+            self.y_dtype_, copy=False
+        )
+
+    def get_metadata(self):
+        return {
+            "classes_": self.classes_,
+            "n_classes_": self.n_classes_,
+            "n_outputs_": self.n_outputs_,
+            "model_n_outputs_": self.model_n_outputs_,
+        }
+
+
+class RegressorTargetEncoder(BaseEstimator, TransformerMixin):
+    """Default target transformer for KerasRegressor.
+    """
+
+    def fit(self, X: np.ndarray) -> "RegressorTargetEncoder":
+        y = X  # rename for clarity, the input is always expected to be a target `y`
+        self.y_dtype_ = y.dtype
+        self.n_outputs_ = 1 if y.ndim == 1 else y.shape[1]
+        self.model_n_outputs_ = 1
+        return self
+
+    def transform(self, X: np.ndarray) -> np.ndarray:
+        y = X
+        n_outputs_ = 1 if y.ndim == 1 else y.shape[1]
+        if n_outputs_ != self.n_outputs_:
```

Comment:
Suggested change:

``` diff
-        n_outputs_ = 1 if y.ndim == 1 else y.shape[1]
-        if n_outputs_ != self.n_outputs_:
+        if y.ndim != self.n_outputs_:
```

Reply:
Also a suggest change above:

``` diff
- self.n_outputs_ = 1 if y.ndim == 1 else y.shape[1]
+ self.n_outputs_ = y.ndim
```

Reply:
I don't think this gives the same result right? `y = np.array([0.1, 0.2, 0.3]).reshape(-1, 1).T` has `y.ndim == 2` and `y.shape[1] == 3` and should correspond to `n_outputs_ == 3`, not `2`. Am I missing something?

Reply:
D'oh, this is a brain-fart on my part. I'm resolving this thread.